react/localState/resolvers/updateItems.js

```diff
@@ -17,7 +24,21 @@ const updateItems = (cartItems, newItems) => {
       })
     )
   }
-  return items.filter(Boolean)
+
+  let itemsRemoved = 0
+  if (itemsToBeRemoved.length > 0) {
+    for (let index = 0; index < items.length; index++) {
```

Comment:
Try using `reduce` or `forEach`, looks more consistent with our codebase (forEach if you really need to mutate `item.optimisticIndex`)